File: src/main/java/um/haberes/core/hexagonal/liquidaciones/codigo/domain/model/Codigo.java
package um.haberes.core.hexagonal.liquidaciones.codigo.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Codigo {

    private Integer codigoId;

    @Builder.Default
    private String nombre = "";

    @Builder.Default
    private Byte docente = 0;

    @Builder.Default
    private Byte noDocente = 0;

    @Builder.Default
    private Byte transferible = 0;

    @Builder.Default
    private Byte incluidoEtec = 0;

    private Long afipConceptoSueldoIdPrimerSemestre;

    private Long afipConceptoSueldoIdSegundoSemestre;
}

File: src/main/java/um/haberes/core/kotlin/model/Codigo.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*

@Entity
@Table
data class Codigo(

    @Id
    var codigoId: Int? = null,
    var nombre: String = "",
    var docente: Byte = 0,
    var noDocente: Byte = 0,
    var transferible: Byte = 0,
    var incluidoEtec: Byte = 0,
    var afipConceptoSueldoIdPrimerSemestre: Long? = null,
    var afipConceptoSueldoIdSegundoSemestre: Long? = null,

    @OneToOne
    @JoinColumn(name = "afipConceptoSueldoIdPrimerSemestre", insertable = false, updatable = false)
    var afipConceptoSueldoPrimerSemestre: AfipConceptoSueldo? = null,

    @OneToOne
    @JoinColumn(name = "afipConceptoSueldoIdSegundoSemestre", insertable = false, updatable = false)
    var afipConceptoSueldoSegundoSemestre: AfipConceptoSueldo? = null

) : Auditable()
